Habitica/src/main/java/com/habitrpg/android/habitica/ui/activities/FullProfileActivity.java

```diff
@@ -416,40 +421,39 @@ public void gotGear(List<ItemData> obj, HabitRPGUser user) {
             }
 
             addEquipmentRow(equipmentTableLayout, i.getKey(), i.getText(), sb.toString());
-        }
 
-        stopAndHideProgress(equipmentProgress);
-        equipmentTableLayout.setVisibility(View.VISIBLE);
+            // Calculate class bonus
+            String itemClass = i.getKlass();
 
-        addAttributeRow(getString(R.string.battle_gear) + ": ", strAttributes, intAttributes, conAttributes, perAttributes, true, false);
-
-        if (!user.getPreferences().isDisableClasses()) {
-            float strClassBonus = 0;
-            float intClassBonus = 0;
-            float conClassBonus = 0;
-            float perClassBonus = 0;
+            if (itemClass.isEmpty()) {
+                continue;
+            }
 
-            switch (user.getStats().get_class()) {
-                case rogue:
-                    strClassBonus = strAttributes * 0.5f;
-                    perClassBonus = perAttributes * 0.5f;
+            switch (itemClass) {
+                case "rogue":
+                    strClassBonus = str_ * 0.5f;
+                    perClassBonus = per_ * 0.5f;
```

Comment:
I'm probably being blind but I can't see where the user's class is taken into account. For example, `case "rogue"` means that the str and per bonuses are calculated for any gear that belongs to the rogue class (I think), but where does it say that the bonus is applied only when the user's class is rogue?